corehq/apps/domain/decorators.py

```diff
@@ -225,6 +234,30 @@ def _inner(request, *args, **kwargs):
     return _inner
 
 
+def check_lockout(fn):
+    @wraps(fn)
+    def _inner(request, *args, **kwargs):
+        username = _get_username_from_request(request)
+        user = CouchUser.get_by_username(username)
+        if user and user.is_web_user() and user.is_locked_out():
+            return HttpResponse(json.dumps({"error": "maximum password attempts exceeded"}),
+                                content_type="application/json",
+                                status=401)
+        else:
+            return fn(request, *args, **kwargs)
+    return _inner
+
+
+def _get_username_from_request(request):
+    auth_type = determine_authtype_from_header(request)
+    username = None
+    if auth_type == 'digest':
```

Comment:
[nip] There are constants for 'digest' and 'basic' in `corehq.apps.domain.auth` that can be used here